optuna/samplers/_gp/sampler.py

```diff
@@ -0,0 +1,209 @@
+
+    def infer_relative_search_space(
+        self, study: Study, trial: FrozenTrial
+    ) -> dict[str, BaseDistribution]:
+        search_space = {}
+        for name, distribution in self._intersection_search_space.calculate(study).items():
+            if distribution.single():
+                continue
+            search_space[name] = distribution
+
+        return search_space
+
+    def sample_relative(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, Any]:
+        self._raise_error_if_multi_objective(study)
+
+        if search_space == {}:
+            return {}
+
+        states = (TrialState.COMPLETE,)
+        trials = study._get_trials(deepcopy=False, states=states, use_cache=True)
+
+        if len(trials) < self._n_startup_trials:
+            return {}
+
+        (
+            internal_search_space,
+            normalized_params,
+        ) = gp_search_space.get_search_space_and_normalized_params(trials, search_space)
+
+        _sign = -1.0 if study.direction == StudyDirection.MINIMIZE else 1.0
+        score_vals = np.array([_sign * cast(float, trial.value) for trial in trials])
+
+        if np.any(~np.isfinite(score_vals)):
+            warnings.warn(
+                "GPSampler cannot handle infinite values. "
+                "We clamp those values to worst/best finite value."
+            )
+
+            finite_score_vals = score_vals[np.isfinite(score_vals)]
+            best_finite_score = np.max(finite_score_vals, initial=0.0)
+            worst_finite_score = np.min(finite_score_vals, initial=0.0)
+
+            score_vals = np.clip(score_vals, worst_finite_score, best_finite_score)
+
+        standarized_score_vals = (score_vals - score_vals.mean()) / max(1e-10, score_vals.std())
+
+        if self._kernel_params_cache is not None and len(
+            self._kernel_params_cache.inverse_squared_lengthscales
+        ) != len(internal_search_space.scale_types):
+            # Clear cache if the search space changes.
+            self._kernel_params_cache = None
+
+        kernel_params = gp.fit_kernel_params(
+            X=normalized_params,
+            Y=standarized_score_vals,
+            is_categorical=(
+                internal_search_space.scale_types == gp_search_space.ScaleType.CATEGORICAL
+            ),
+            log_prior=self._log_prior,
+            minimum_noise=self._minimum_noise,
+            initial_kernel_params=self._kernel_params_cache,
+        )
+        self._kernel_params_cache = kernel_params
+
+        acqf_params = acqf.create_acqf_params(
+            kernel_params=kernel_params,
+            search_space=internal_search_space,
+            X=normalized_params,
+            Y=standarized_score_vals,
+        )
+
+        normalized_param, _ = optim.optimize_acqf_sample(
+            acqf_params,
+            n_samples=self._optimize_n_samples,
+            seed=self._rng.rng.random_integers(0, np.iinfo(np.int32).max),
```

Comment:
`random_integers` is already deprecated in the latest version, so let's use `randint`.
```suggestion
            seed=self._rng.rng.random_int(np.iinfo(np.int32).max),
```

Reply:
```suggestion
            seed=self._rng.rng.randint(np.iinfo(np.int32).max),
```